vm/src/obj/objlist.rs

```diff
@@ -484,17 +498,17 @@ impl PyList {
 
     #[pymethod(name = "__imul__")]
     fn imul(zelf: PyRef<Self>, counter: isize) -> PyRef<Self> {
-        let new_elements = sequence::seq_mul(&zelf.borrow_sequence(), counter)
-            .cloned()
-            .collect();
-        zelf.elements.replace(new_elements);
-        zelf
+        let elements = zelf.borrow_elements_mut();
+        let new_elements: Vec<PyObjectRef> =
+            sequence::seq_mul(&elements, counter).cloned().collect();
+        PyList::_replace(elements, new_elements);
+        zelf.clone()
     }
 
     #[pymethod]
     fn count(&self, needle: PyObjectRef, vm: &VirtualMachine) -> PyResult<usize> {
         let mut count: usize = 0;
-        for element in self.elements.borrow().iter() {
+        for element in self.borrow_elements().clone().iter() {
```

Comment:
It'd be really nice if we could somehow detect if something tries to lock while we already have one and then give it a clone, but I think any solution to that would violate rust's aliasing rules.

Reply:
The big question is how expensive is the `clone`. I hope that this is very quick because of the `Arc`.